### ISSUE TITLE
Natron 2.3.14 doesn't start

### ISSUE THREAD

**Issue by D4N13L1980** [NatronGitHub/Natron#317]
### Problem

Natron2 doesn't start.

**Expected behavior:** 

Start Natron2

**Actual behavior:** 

Natron2 doesn't start

 
 * If your problem is a crash in an official release/snapshot, please include verbose output from the application
 from a terminal if possible.  If you also submitted a crash report, indicate the CrashID if possible.

Natron Version 2.3.14
Copyright (C) 2013-2018 INRIA and Alexandre Gauthier-Foichat
>>>Use the --help or -h option to print usage.<<<
Caught segmentation fault (SIGSEGV) from thread Main(0x382d290), faulty address is 0x7f3a42be534a from 0x22
Backtrace:
[Frame 1]: /opt/amdgpu-pro/lib/x86_64-linux-gnu/libGL.so.1(+0x5734a) [0x7f3a42be534a]
[Frame 2]: /opt/amdgpu-pro/lib/x86_64-linux-gnu/libGL.so.1(+0x5734a) [0x7f3a42be534a]
[Frame 3]: /usr/lib/x86_64-linux-gnu/dri/amdgpu_dri.so(+0xc75748) [0x7f3a3a856748]
[Frame 4]: /usr/lib/x86_64-linux-gnu/dri/amdgpu_dri.so(+0x137fa74) [0x7f3a3af60a74]
[Frame 5]: /usr/lib/x86_64-linux-gnu/dri/amdgpu_dri.so(+0x63cdeb) [0x7f3a3a21ddeb]
[Frame 6]: /usr/lib/x86_64-linux-gnu/dri/amdgpu_dri.so(+0xb7232a) [0x7f3a3a75332a]
[Frame 7]: /usr/lib/x86_64-linux-gnu/dri/amdgpu_dri.so(+0x109fada) [0x7f3a3ac80ada]
[Frame 8]: /usr/lib/x86_64-linux-gnu/dri/amdgpu_dri.so(+0x10a11cc) [0x7f3a3ac821cc]
[Frame 9]: /usr/lib/x86_64-linux-gnu/dri/amdgpu_dri.so(+0x10a14cd) [0x7f3a3ac824cd]
[Frame 10]: /opt/amdgpu-pro/lib/x86_64-linux-gnu/libGL.so.1(+0x58cd4) [0x7f3a42be6cd4]
[Frame 11]: /opt/amdgpu-pro/lib/x86_64-linux-gnu/libGL.so.1(glXCreateContextAttribsARB+0x1e8) [0x7f3a42bdab28]
[Frame 12]: /home/d4n13l/Natron2/bin/Natron(_ZN6Natron22OSGLContext_x11Private16createContextGLXEPNS_20OSGLContext_glx_dataERKNS_17FramebufferConfigEiibiPKNS_15OSGLContext_x11E+0x1c5) [0xdd45c5]
[Frame 13]: /home/d4n13l/Natron2/bin/Natron(_ZN6Natron15OSGLContext_x11C2ERKNS_17FramebufferConfigEiibRKNS_12GLRendererIDEPKS0_+0xd7) [0xdd4aa7]
[Frame 14]: /home/d4n13l/Natron2/bin/Natron(_ZN6Natron15OSGLContext_x1111getGPUInfosERNSt7__cxx114listINS_18OpenGLRendererInfoESaIS3_EEE+0x12e2) [0xdd5e02]
[Frame 15]: /home/d4n13l/Natron2/bin/Natron(_ZN6Natron10AppManager12loadInternalERKNS_6CLArgsE+0x49c) [0xb5ccac]

### Versions

Natron Version 2.3.14
Release: Natron-2.3.14-Linux-x86_64bit.tgz

* **OS version:** 
Ubuntu 18.04.1 LTS

**Comment by devernay** [comment]
- make sure LD_LIBRARY_PATH is not set (`echo $LD_LIBRARY_PATH`)
- check that `ldd /home/d4n13l/Natron2/bin/Natron` does not point to system-installed libraries (give the output of this command here if you don't know)

**Comment by D4N13L1980** [comment]
I don't know so I give you the output^^
	
linux-vdso.so.1 (0x00007ffe02db1000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007f80f5b3d000)
	libpython2.7.so.1.0 => /home/d4n13l/Natron2/bin/../lib/libpython2.7.so.1.0 (0x00007f80f5719000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f80f5515000)
	libutil.so.1 => /lib/x86_64-linux-gnu/libutil.so.1 (0x00007f80f5312000)
	libboost_serialization.so.1.67.0 => /home/d4n13l/Natron2/bin/../lib/libboost_serialization.so.1.67.0 (0x00007f80f50cb000)
	libexpat.so.1 => /home/d4n13l/Natron2/bin/../lib/libexpat.so.1 (0x00007f80f4e9a000)
	libpyside-python2.7.so.1.2 => /home/d4n13l/Natron2/bin/../lib/libpyside-python2.7.so.1.2 (0x00007f80f4c6c000)
	libshiboken-python2.7.so.1.2 => /home/d4n13l/Natron2/bin/../lib/libshiboken-python2.7.so.1.2 (0x00007f80f4a3d000)
	libGLU.so.1 => /usr/lib/x86_64-linux-gnu/libGLU.so.1 (0x00007f80f47ce000)
	libpixman-1.so.0 => /home/d4n13l/Natron2/bin/../lib/libpixman-1.so.0 (0x00007f80f452b000)
	libfontconfig.so.1 => /home/d4n13l/Natron2/bin/../lib/libfontconfig.so.1 (0x00007f80f42ea000)
	libfreetype.so.6 => /home/d4n13l/Natron2/bin/../lib/libfreetype.so.6 (0x00007f80f4037000)
	libQtOpenGL.so.4 => /home/d4n13l/Natron2/bin/../lib/libQtOpenGL.so.4 (0x00007f80f3d3d000)
	libQtGui.so.4 => /home/d4n13l/Natron2/bin/../lib/libQtGui.so.4 (0x00007f80f30c3000)
	libQtNetwork.so.4 => /home/d4n13l/Natron2/bin/../lib/libQtNetwork.so.4 (0x00007f80f2d90000)
	libQtCore.so.4 => /home/d4n13l/Natron2/bin/../lib/libQtCore.so.4 (0x00007f80f28ad000)
	libGL.so.1 => /opt/amdgpu-pro/lib/x86_64-linux-gnu/libGL.so.1 (0x00007f80f259f000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f80f2380000)
	libstdc++.so.6 => /home/d4n13l/Natron2/bin/../lib/libstdc++.so.6 (0x00007f80f1fe8000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f80f1c4a000)
	libgomp.so.1 => /home/d4n13l/Natron2/bin/../lib/libgomp.so.1 (0x00007f80f1a1f000)
	libgcc_s.so.1 => /home/d4n13l/Natron2/bin/../lib/libgcc_s.so.1 (0x00007f80f1809000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f80f1418000)
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f80f11f0000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f80f5e75000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f80f0fe8000)
	libbz2.so.1 => /home/d4n13l/Natron2/bin/../lib/libbz2.so.1 (0x00007f80f0dd8000)
	libpng16.so.16 => /home/d4n13l/Natron2/bin/../lib/libpng16.so.16 (0x00007f80f0ba4000)
	libz.so.1 => /home/d4n13l/Natron2/bin/../lib/libz.so.1 (0x00007f80f098e000)
	libuuid.so.1 => /home/d4n13l/Natron2/bin/../lib/libuuid.so.1 (0x00007f80f078a000)
	libXi.so.6 => /usr/lib/x86_64-linux-gnu/libXi.so.6 (0x00007f80f057a000)
	libXrender.so.1 => /usr/lib/x86_64-linux-gnu/libXrender.so.1 (0x00007f80f0370000)
	libXrandr.so.2 => /usr/lib/x86_64-linux-gnu/libXrandr.so.2 (0x00007f80f0165000)
	libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007f80eff5f000)
	libXcursor.so.1 => /usr/lib/x86_64-linux-gnu/libXcursor.so.1 (0x00007f80efd55000)
	libXinerama.so.1 => /usr/lib/x86_64-linux-gnu/libXinerama.so.1 (0x00007f80efb52000)
	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007f80ef940000)
	libgthread-2.0.so.0 => /home/d4n13l/Natron2/bin/../lib/libgthread-2.0.so.0 (0x00007f80ef73f000)
	libglib-2.0.so.0 => /home/d4n13l/Natron2/bin/../lib/libglib-2.0.so.0 (0x00007f80ef42d000)
	libSM.so.6 => /usr/lib/x86_64-linux-gnu/libSM.so.6 (0x00007f80ef225000)
	libICE.so.6 => /usr/lib/x86_64-linux-gnu/libICE.so.6 (0x00007f80ef00a000)
	libssl.so.1.0.0 => /home/d4n13l/Natron2/bin/../lib/libssl.so.1.0.0 (0x00007f80eed9c000)
	libcrypto.so.1.0.0 => /home/d4n13l/Natron2/bin/../lib/libcrypto.so.1.0.0 (0x00007f80ee94d000)
	libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007f80ee74a000)
	libX11-xcb.so.1 => /usr/lib/x86_64-linux-gnu/libX11-xcb.so.1 (0x00007f80ee548000)
	libxcb-glx.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-glx.so.0 (0x00007f80ee32d000)
	libxcb-dri3.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri3.so.0 (0x00007f80ee129000)
	libxcb-present.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-present.so.0 (0x00007f80edf26000)
	libxcb-dri2.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri2.so.0 (0x00007f80edd21000)
	libXxf86vm.so.1 => /usr/lib/x86_64-linux-gnu/libXxf86vm.so.1 (0x00007f80edb1b000)
	libxshmfence.so.1 => /usr/lib/x86_64-linux-gnu/libxshmfence.so.1 (0x00007f80ed919000)
	libxcb-sync.so.1 => /usr/lib/x86_64-linux-gnu/libxcb-sync.so.1 (0x00007f80ed712000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007f80ed50e000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f80ed308000)
	libpcre.so.1 => /home/d4n13l/Natron2/bin/../lib/libpcre.so.1 (0x00007f80ed0c5000)
	libbsd.so.0 => /lib/x86_64-linux-gnu/libbsd.so.0 (0x00007f80eceb0000)

**Comment by devernay** [comment]
can you check also `ldd /opt/amdgpu-pro/lib/x86_64-linux-gnu/libGL.so.1`?
some openGL implementations depend on the system-installed libstdc++, and it may break if it is a different version than the one shipped with Natron. If this is the case, I don't know how to handle this, except by compiling Natron on your system with the same compiler used for libstdc++

**Comment by D4N13L1980** [comment]
output of: ldd /opt/amdgpu-pro/lib/x86_64-linux-gnu/libGL.so.1

linux-vdso.so.1 (0x00007fff2870a000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fada4fdb000)
	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007fada4dc9000)
	libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007fada4bc6000)
	libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007fada49c0000)
	libX11-xcb.so.1 => /usr/lib/x86_64-linux-gnu/libX11-xcb.so.1 (0x00007fada47be000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007fada4486000)
	libxcb-glx.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-glx.so.0 (0x00007fada426b000)
	libxcb-dri3.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri3.so.0 (0x00007fada4067000)
	libxcb-present.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-present.so.0 (0x00007fada3e64000)
	libxcb-dri2.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-dri2.so.0 (0x00007fada3c5f000)
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007fada3a37000)
	libXxf86vm.so.1 => /usr/lib/x86_64-linux-gnu/libXxf86vm.so.1 (0x00007fada3831000)
	libxshmfence.so.1 => /usr/lib/x86_64-linux-gnu/libxshmfence.so.1 (0x00007fada362f000)
	libxcb-sync.so.1 => /usr/lib/x86_64-linux-gnu/libxcb-sync.so.1 (0x00007fada3428000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fada308a000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fada2c99000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fada5508000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fada2a95000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007fada2891000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007fada268b000)
	libbsd.so.0 => /lib/x86_64-linux-gnu/libbsd.so.0 (0x00007fada2476000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fada226e000)

**Comment by threedslider** [comment]
What is your graphic card  ? do it as : ubuntu-drivers devices

Then : sudo ubuntu-drivers autoinstall if it works

Or

You will need to compile this Natron by source code on your Ubuntu and it will resolve i think.

Link : [Instruction from sources on Linux ](https://github.com/NatronGitHub/Natron/blob/RB-2.3/INSTALL_LINUX.md)

**Comment by D4N13L1980** [comment]
Thanks.
My graphic card: Radeon PRO WX 8200

After: ~$ sudo ubuntu-drivers autoinstall
No drivers found for automatic installation.

My Drivers are well setup and working.

So I have to compile I guess?

**Comment by threedslider** [comment]
See if you can update for driver last version in [amd support ](https://www.amd.com/en/support) ?

Or for technical on how to install it : [How to Install The Latest AMD Radeon Drivers on Ubuntu 18.04 Bionic Beaver Linux 
](https://linuxconfig.org/how-to-install-the-latest-amd-radeon-drivers-on-ubuntu-18-04-bionic-beaver-linux)

**Comment by dkudos** [comment]
> Thanks.
> My graphic card: Radeon PRO WX 8200
> 
> After: ~$ sudo ubuntu-drivers autoinstall
> No drivers found for automatic installation.
> 
> My Drivers are well setup and working.
> 
> So I have to compile I guess?

Did you get it to work or are you still stuck?

**Comment by ebarranco** [comment]
Hi , I'm in manjaro Linux KDE 18.0.2 and Natron portable or installed doesn't start here.
`Natron Version 2.3.14
Copyright (C) 2013-2018 INRIA and Alexandre Gauthier-Foichat
>>>Use the --help or -h option to print usage.<<<
couldn't open library /home/ebarranco/Natron2/Plugins/OFX/Natron/Arena.ofx.bundle/Contents/Linux-x86-64/Arena.ofx because /usr/lib/libcairo.so.2: undefined 
Caught segmentation fault (SIGSEGV) from thread Main(0x322f9d0), faulty address is 0x366a0 from 0x366a0
`

**Comment by jeinstei** [comment]
> Hi , I'm in manjaro Linux KDE 18.0.2 and Natron portable or installed doesn't start here.
> `Natron Version 2.3.14
> Copyright (C) 2013-2018 INRIA and Alexandre Gauthier-Foichat
> 
> > > > Use the --help or -h option to print usage.<<<
> > > > couldn't open library /home/ebarranco/Natron2/Plugins/OFX/Natron/Arena.ofx.bundle/Contents/Linux-x86-64/Arena.ofx because /usr/lib/libcairo.so.2: undefined
> > > > Caught segmentation fault (SIGSEGV) from thread Main(0x322f9d0), faulty address is 0x366a0 from 0x366a0
> > > > `

Not to be "that guy", but this sounds like it should be a different issue as it is referencing a different library. @ebarranco , does your system have libcairo even installed? I'm not a developer on Natron, so I have no idea if libcairo should be packaged with Natron as well.

**Comment by dilom** [comment]
> Hi , I'm in manjaro Linux KDE 18.0.2 and Natron portable or installed doesn't start here.
> `Natron Version 2.3.14
> Copyright (C) 2013-2018 INRIA and Alexandre Gauthier-Foichat
> 
> > > > Use the --help or -h option to print usage.<<<
> > > > couldn't open library /home/ebarranco/Natron2/Plugins/OFX/Natron/Arena.ofx.bundle/Contents/Linux-x86-64/Arena.ofx because /usr/lib/libcairo.so.2: undefined
> > > > Caught segmentation fault (SIGSEGV) from thread Main(0x322f9d0), faulty address is 0x366a0 from 0x366a0
> > > > `

I have the exact same problem and error message. I did create a new issue, but it was closed and I got referred to continue the discussion here. So any help would be appreciated

**Comment by devernay** [comment]
Actually, this is probably the same error.
My guess is that the plugins (eg `/home/ebarranco/Natron2/Plugins/OFX/Natron/Arena.ofx.bundle/Contents/Linux-x86-64/Arena.ofx`) do not have the ld run-path set correctly.
In a terminal, try setting `LD_LIBRARY_PATH` to where Natron stores its libraries (eg `/home/ebarranco/Natron2/lib` and start natron from this terminal (eg `/home/ebarranco/Natron2/bin/Natron`)

**Comment by dilom** [comment]
Just tried it, same error message. Seems to be something related. I just installed the application from the site, also tried to do download and unpack manually. It seems some path is not set correctly.

So if I disable/delete the Arena plugin will the program run? If this is not a core plugin, I can live with that for the time being.

**Comment by jeinstei** [comment]
Is there any way a plug in could hard code a library path?

**Comment by ebarranco** [comment]
@devernay  I tried it too. No change.  maybe is important to say that I already had installed and used Natron when stopped working. Then I did uninstall and download both installer and portable with the same result. Later I installed Linux in two new computers with the same result when a try install Natron. Now I'm using flatpak, it's working but the version is outdated.

**Comment by dilom** [comment]
So, I moved the Arena plugin out of the plugin directory and Natron starts fine. I am new to the application so I have no idea what functionality I lost but now at least I can learn and test the software.

**Comment by ebarranco** [comment]
@dilom They are a bunch of nodes from the star (icon) menu I think. Yes,  I delete that and natron starts fine, just doesn't have that menu anymore and  I have used a couple of those before. I tried to compile the plugin but fail.

**Comment by ebarranco** [comment]
I did report that the Flatpak version was outdated. They fixed it and now I updated Natron and It has the same problem that the official installer and portable and don't start the program.

**Comment by threedslider** [comment]
Ok, this is more problem for Natron runs with this software, already i am busy right now and i work on Natron with 3D Camera, so after that maybe i will work to make all in one installation cause it will facilitate for you on Linux system to run it. I will learn shell command to put in correct way for easy installation (automatic set env, path and all in one folder with sub-folder for executable and libraries)

**Comment by Sunderland93** [comment]
Flatpak version works fine on Debian 9, but failed on Ubuntu 18.04 with same error....

**Comment by Sunderland93** [comment]
Flatpak version fixed now

**Comment by ebarranco** [comment]
Thanks @Sunderland93

**Comment by rodlie** [comment]
This is a issue with libstdc++ distributed with Natron.

@D4N13L1980 : If you have equal or higher gcc on your system than what is included in Natron remove libstdc++/libgomp/libgcc from Natron2/lib.

**Comment by devernay** [comment]
Duplicate of #342

**Comment by gumbo-cms** [comment]
> This is a issue with libstdc++ distributed with Natron.
> 
> @D4N13L1980 : If you have equal or higher gcc on your system than what is included in Natron remove libstdc++/libgomp/libgcc from Natron2/lib.

I'm on Fedora 31 64 bits and this solves the problem!  Thank you

**Comment by rodlie** [comment]
@gumbo-cms : The latest 2.3.15 RC's for Linux fixes the issue with libstdc++.

**Comment by gumbo-cms** [comment]
> @gumbo-cms : The latest 2.3.15 RC's for Linux fixes the issue with libstdc++.

Actually, this morning I've switched to Natron-RB-2.3-202004080542-35a821c-Linux-64-no-installer.tar.xz and so far so good. Using the stable version I've quickly bumped into an error message trying to read an SVG file, then switching to the latest RC version solved the issue and everything is fine for now.
